### PR TITLE
*: Convert expression to pbExpr and implement builtin function

### PR DESCRIPTION
**Pull request by zimulala**

### CODE REVIEW

File: executor/new_executor.go [hunk 1]
```diff
@@ -788,6 +824,64 @@ func (b *executorBuilder) columnToPBExpr(client kv.Client, column *expression.Co
...
+	if !client.SupportRequestType(kv.ReqTypeSelect, int64(tipb.ExprType_In)) {
+		return nil
+	}
+
+	pbExpr := b.newExprToPBExpr(client, expr.Args[0], tbl)
+	if pbExpr == nil {
+		return nil
+	}
+	var listExpr *tipb.Expr
+	if len(expr.Args[1:]) == 1 {
```
Comment (coocood) [inline executor/new_executor.go]: Why not `len(expr.Args) == 2` ?

File: executor/new_executor.go [hunk 2]
```diff
@@ -810,9 +904,30 @@ func (b *executorBuilder) scalarFuncToPBExpr(client kv.Client, expr *expression.
 		tp = tipb.ExprType_And
 	case ast.Or:
 		tp = tipb.ExprType_Or
-	// It's the operation for unary operator.
 	case ast.UnaryNot:
-		tp = tipb.ExprType_Not
+		return b.notToPBExpr(client, expr, tbl)
+	case ast.In:
+		return b.inToPBExpr(client, expr, tbl)
+	// Only patterns like 'abc', '%abc', 'abc%', '%abc%' can be converted to *tipb.Expr for now.
```
Comment (coocood) [inline executor/new_executor.go]: move this comment below `case ast.Like`.

File: executor/new_executor.go [hunk 3]
```diff
@@ -788,6 +824,64 @@ func (b *executorBuilder) columnToPBExpr(client kv.Client, column *expression.Co
...
+		return nil
+	}
+
+	pbExpr := b.newExprToPBExpr(client, expr.Args[0], tbl)
+	if pbExpr == nil {
+		return nil
+	}
+	var listExpr *tipb.Expr
+	if len(expr.Args[1:]) == 1 {
+		// Args[1] is a select subquery, or it's a list and its length is 1.
```
Comment (hanfei19910905) [inline executor/new_executor.go]: args[1] will never be a select subquery.

**Comment by hanfei19910905** [comment]
implement the builtin function and add test to executor_test.

**Comment by coocood** [comment]
LGTM

**Comment by zimulala** [comment]
PTAL @hanfei19910905 @coocood @shenli

**Comment by hanfei19910905** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**